R/utilities-mass-balance.R

```diff
@@ -357,3 +318,162 @@ getMassBalanceDataMapping <- function(plotType) {
   )
   return(dataMapping)
 }
+
+#' @title getApplicationResults
+#' @description Get a data.frame of application results corresponding to total drug mass as a function of time.
+#' Accounts for infusion cases in which infusion time is not `NULL` by including intermediate time steps
+#' @param applications list of `Application` objects queried by the method `simulation$allApplicationsFor()`
+#' @param infusionSteps number of time steps to discretize the infusion time
+#' @return A data.frame that includes `time`, `drugMass` and `totalDrugMass` as variables
+#' @import ospsuite
+#' @import ospsuite.utils
+#' @import dplyr
+#' @keywords internal
+getApplicationResults <- function(applications, infusionSteps = 20) {
```

Comment:
let's make our life simple and calculate `drugmass(t)=sum_of_drugmass_of_all_applications_starting at Time<=t`
If we start to account for the drugmass which was **effectively** applied (taking the infusion rate into account) - then we would e.g. need to do the same for tablets, which are also dissolved with time - which in general cannot be even estimated apriori without the simulation.

And for infusion, the current implementation is wrong anyways.
I tried it for a single infusion administration in a simulation (infusion time 10 minutes) and got this:
```
> getApplicationResults(applications)
         time  drugMass totalDrugMass
1   0.0000000 0.1560549     0.1560549
2   0.5263158 0.1560549     0.3121099
3   1.0526316 0.1560549     0.4681648
4   1.5789474 0.1560549     0.6242197
5   2.1052632 0.1560549     0.7802747
6   2.6315789 0.1560549     0.9363296
7   3.1578947 0.1560549     1.0923845
8   3.6842105 0.1560549     1.2484395
9   4.2105263 0.1560549     1.4044944
10  4.7368421 0.1560549     1.5605493
11  5.2631579 0.1560549     1.7166042
12  5.7894737 0.1560549     1.8726592
13  6.3157895 0.1560549     2.0287141
14  6.8421053 0.1560549     2.1847690
15  7.3684211 0.1560549     2.3408240
16  7.8947368 0.1560549     2.4968789
17  8.4210526 0.1560549     2.6529338
18  8.9473684 0.1560549     2.8089888
19  9.4736842 0.1560549     2.9650437
20 10.0000000 0.1560549     3.1210986
```